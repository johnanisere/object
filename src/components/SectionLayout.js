import React from "react";
import PropTypes from "prop-types";

import SectionLayout from "../blocks/sectionLayout";

const App = ({ title, children, dark, loading }) => (
  <SectionLayout loading={loading}>
    <SectionLayout.Header dark={dark}>{title}</SectionLayout.Header>
    <div className="child">{children}</div>
  </SectionLayout>
);

App.propTypes = {
  dark: PropTypes.bool,
  loading: PropTypes.bool,
  children: PropTypes.node.isRequired
};

export default App;
