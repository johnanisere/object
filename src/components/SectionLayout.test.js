
import React from "react";
import { shallow } from "enzyme";

import SectionLayout from "./SectionLayout"
import SectionLayoutConatiner from "../blocks/sectionLayout";

it("SectionLayout component exists", () => {
    const wrapper = shallow(<SectionLayout><div /></SectionLayout>);
    expect(wrapper.exists()).toBe(true);
});

it("SectionLayout component renders", () => {
    const wrapper = shallow(<SectionLayout><div /></SectionLayout>);
    expect(wrapper).toMatchSnapshot();
});

it(`SectionLayout should contain "SectionLayoutContainer",  "span","" `, () => {
    const wrapper = shallow(<SectionLayout><div /></SectionLayout>);

    expect(wrapper.find('div')).toHaveLength(2);
    expect(wrapper.find(SectionLayoutConatiner)).toHaveLength(1);
    expect(wrapper.find(SectionLayoutConatiner.Header)).toHaveLength(1);
});