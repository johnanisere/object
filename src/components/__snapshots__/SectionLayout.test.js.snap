// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionLayout component renders 1`] = `ShallowWrapper {}`;
